app/src/main/java/com/hedvig/app/feature/offer/ui/changestartdate/ChangeDateBottomSheet.kt

```diff
@@ -77,7 +82,8 @@ class ChangeDateBottomSheet : BottomSheetDialogFragment() {
             currentInsurerDisplayName = inception.currentInsurer?.displayName,
             startDate = inception.startDate,
             switchable = inception.currentInsurer?.switchable ?: false,
-            datePickerListener = {
+            datePickerListener = datePickerListener@{
```

Comment:
What made this check required? 🤔

Reply:
Couldn't do a local return inside the lambda without explicitly naming it like this.
The name it had by itself was `bind` so I could do `return@bind` but that one felt confusing, you have to go and look for what is named bind which is some lines further up.
Naming it like this makes it super clear to understand what you are doing a local return on.
But anyway, `return@bind` is an option too if you think this solution is too weird. Labling an anonymous function like this is definitely not something you do every day. Maybe even something you shouldn't? I don't have any strong personal opinions on this or examples of when it's not a good idea?

Reply:
I meant why we needed to check `shouldOpenDatePicker` 😬

Reply:
Oh 🤦‍♂️
So what happens if we do not do this check: 
After you click on Save button, it gets disabled thus you can not click it again while the result is loading. And this does take a solid half a second or more most of the times. But in the meantime, you can actually click on the date picker again. And then you get the date dialog, and you're confused since even if you do make a change, when the save result comes back, the dialog will just close on your face and the bottom sheet will be redacted. 
So I thought just like we disable the button, we should also disable the date picker.

Reply:
Sensible! But perhaps we could do it in the same way as we do for the dismissibility of the dialog? 😬

Reply:
I started out by trying to do it this way. But we create this view programmatically inside the .Inceptions case of the view state, there is no reference to it outside of there. And setting its parent view as non clickable, or disabled etc has no effect on the click listener. The only way I found to fix it is this way, maybe I am missing something however.

Reply:
Fair enough, then this is probably the best approach indeed.